linkerd/app/inbound/src/lib.rs

```diff
@@ -119,11 +118,21 @@ impl Inbound<()> {
                 ..
             } = config.proxy.connect;
 
+            #[derive(Debug, thiserror::Error)]
+            #[error("inbound connection must not target port {0}")]
+            struct Loop(u16);
+
             svc::stack(transport::ConnectTcp::new(*keepalive))
-                .push_map_target(|t: T| Remote(ServerAddr(([127, 0, 0, 1], t.param()).into())))
                 // Limits the time we wait for a connection to be established.
                 .push_connect_timeout(*timeout)
-                .push(svc::stack::BoxFuture::layer())
```

Comment:
This `BoxFuture` layer is redundant -- there's already a `BoxFuture` in the HTTP router where it's actually needed.